pkg/core/rest/errors/error_handler.go

```diff
@@ -145,6 +146,12 @@ func HandleError(ctx context.Context, response *restful.Response, err error, tit
 			Title:  title,
 			Detail: err.Error(),
 		}
+	case errors.Is(err, &admin.KDSTransportError{}):
+		kumaErr = &types.Error{
+			Status: 400,
```

Comment:
Is 400 really the right status here? There's no user error afaiu. I think if we want to use 4xx we should use another status code (potentially a non standard one)

Reply:
I'm fine either with 400 or with non-standard 4xx error. It both indicates user-facing error which is technically not correct

Reply:
It's tricky to pick a good status code here, but I was thinking "who can do something to resolve the problem" and in this case, it's a user. So it's a bit of a stretch, but I think it's acceptable to treat it as a user error

Reply:
Do we also need to consider how the GUI responds to this error?

Reply:
Not really, it's just going to display the error text, we already have a similar 400 error for StreamNotConnected https://github.com/kumahq/kuma/blob/9146696f3a05a8c0bc70c0b711a1b1ac8e75914e/pkg/core/rest/errors/error_handler.go#L131